django_lightweight_queue/machine_types.py

```diff
@@ -0,0 +1,101 @@
+from django.utils.functional import cached_property
+
+from . import app_settings
+from .cron_scheduler import CRON_QUEUE_NAME
+
+
+class Machine(object):
+    """
+    Dummy machine class to contain documentation.
+
+    Implementations may extend this class if desired, though this
+    is not required.
+    """
+
+    @property
+    def run_cron(self):
+        """
+        Returns a `bool` for whether or not a runner on this machine should
+        run the cron queue.
+        """
+        raise NotImplementedError()
+
+    @property
+    def worker_names(self):
+        """
+        Returns a list of tuples of (queue_name, worker_num) for the workers
+        which should run on this machine. Worker numbers start at 1.
+
+        Implemetations should be efficient even if this is called several times.
+        """
+        raise NotImplementedError()
+
+
+def get_workers_names(machine_number, machine_count, only_queue):
```

Comment:
It's only ever used as part of a `PooledMachine`. I nearly made it a method, but it seemed nicer to keep it separate and pass in the requisite member variables.

(It's also somewhat coupled to these classes by the shape of its return value)